src/layout/Main/index.tsx

```diff
@@ -138,6 +149,31 @@ function MainLayout({ children }: PropsWithChildren) {
   const connectWalletModal = useConnectWalletModal();
   const [searchParams, setSearchParams] = useSearchParams();
 
+  const handleModalClose = useCallback(() => {
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set(CHAIN_NAME_SEARCH_PARAM, DefaultChainName);
+    setSearchParams(newParams);
+    window.location.reload();
+  }, [searchParams, setSearchParams]);
+
+  const { isOpen, open } = useInvalidPathModal({
+    onReturnClick: handleModalClose,
+  });
+
+  const { paramName, isValidChain } = useMemo(() => {
+    const searchName = searchParams.get(CHAIN_NAME_SEARCH_PARAM);
+    const isSupported = DefaultChain.some(
+      (supportChain) => supportChain.chainName === searchName,
+    );
+    return { paramName: searchName, isValidChain: isSupported };
+  }, [searchParams]);
+
+  useEffect(() => {
+    if (paramName && !isValidChain) {
+      open();
+    }
+  }, [isOpen, isValidChain, open, paramName, searchParams]);
```

Comment:
```suggestion
  useEffect(() => {
    open(!!(paramName && !isValidChain));
  }, [isOpen, isValidChain, open, paramName, searchParams]);
```

what do you think about closing modal here without reloading app in `handleModalClose`?
if we'd like to show invalid URL modal while being connected too, we could add `isOpen` check to useEffect (L178)

Reply:
currently, it seems somewhat awkward for me:

https://github.com/user-attachments/assets/f4d9a666-37ed-4c5d-b763-42845b3d2b8d

it corrects chain name in params before the modal shows up.
and closing modal is followed by some flickers(?).

I think this way seems more natural without reloading:

https://github.com/user-attachments/assets/e9e9996c-bd13-4e81-bac8-b8e4d52011f2

Reply:
[d66b39b](https://github.com/dezswap/dezswap-web-app/pull/248/commits/d66b39b2ed24b6adb56c9079b3d37ea4b9e6956b) is updated

Reply:
@winnieworld Thank you. btw, my point was about avoiding `window.location.reload()` call. sorry I might confuse you.

Reply:
[6a6889b](https://github.com/dezswap/dezswap-web-app/pull/248/commits/6a6889b1eb40b32fbc10f3423e96d9b08114d36f) is updated